### PR TITLE
Refactor the policy output code and move png and csv download buttons to the right location

### LINKED ISSUE #759: Chart `Download` button/dropdown is in the left pane
Instead of by the chart
image

### PR DESCRIPTION
**Pull request by abhcs**
## The design of `pages/policy/output`

`PolicyOutput` is the top-level component with the following functions:

- fetch the impact data for the current policy and impact type from the back end
- display the impact data for the current policy and impact type as a chart
- display buttons for performing actions
  - download png, csv
  - copy link
  - share on social media (Twitter, LinkedIn, etc)
- display appropriate messages in the following cases
  - no policy is selected
  - the front end is waiting for the back end
  - the back end returns an error

The `impact` object returned by the back end has a hierarchical structure describing the impacts for various impact types. For instance, `impact.budget.budgetary_impact` and `impact.poverty.poverty` contain the budgetary and poverty impact data respectively. The impact does not need to be refetched when switching between these impact types. `PolicyOutput` fetches the `impact` object and handles errors. If there are no errors, then `PolicyOutput` displays the charts and action buttons. Since each impact type has a unique chart, `PolicyOutput` delegates the chart display functionality to low-level components for each impact type, e.g., `BudgetaryImpact` and `PovertyImpact`. In addition to displaying the chart for the current policy and impact type, the low-level components are also responsible for

- generating the csv data, which depends on the impact type
- displaying the download png and csv buttons

`PolicyOutput` retains the responsibility for displaying the other action buttons.

The low-level component `CliffOutput` is an exception. It is responsible for fetching its own impact object from a different source than the other impact types. Aside from this additional responsibility, `CliffOutput` is similar to other low-level components.

## Improving the design

In the current design, the separation of concerns is a little blurred. Many components have too many responsibilities assigned to them. We break `PolicyOutput` into smaller components for

- fetching the impact data for all impact types including cliff
- displaying error and wait messages
- displaying charts by delegating to low-level components
- displaying action buttons (this component is also used by the household calculator)

For all impact types including cliff, we require the low-level components to be _uniformly_ responsible _only_ for generating the chart and providing a function that generates the csv data.

There is a lot of repetition in the charts code -- we refactor this code to make it less repetitive. For instance, `IntraDecileImpact` and `IntraWealthDecileImpact` have essentially the same code for creating the plot, but the code is repeated in the two files. Thus, improvements made to the `IntraDecileImpact` plot wrt legends had not propagated to `IntraWealthDecileImpact`. We eliminate this kind of repetition in the code.

We fix #759 by moving the download png and csv actions next to the buttons for sharing results.

## Screenshots

Screenshot 2023-12-25 at 10 46 13 PM

## Tests

Several dozen configurations have been tested but more testing is needed for a change of this magnitude.

### CODE REVIEW

**Comment by anth-volk** [comment]
Before proceeding on this, could you output a screenshot of the same image above, but using the refundable CTC? This is a dummy check, since no one should be worse off with it.

**Comment by abhcs** [comment]
Do you have a url for the policy? Is it [this one](https://policyengine.org/us/policy?focus=policyOutput.intraDecileImpact&region=us&timePeriod=2023&baseline=2&reform=3170)?

**Comment by anth-volk** [comment]
It is that one, yeah

**Comment by abhcs** [comment]
Screenshot 2023-12-26 at 2 57 31 PM

**Comment by MaxGhenis** [comment]
Looks good! Not sure about the sliver but it's not a front-end issue, filed https://github.com/PolicyEngine/policyengine-us/issues/3536

File: src/pages/policy/output/Display.jsx [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+              throw new Error("Something went wrong");
+            }
+          });
+        });
+      }, 1000);
+    }
+  }, [preparingForScreenshot, takeScreenShot]);
+
+  const urlParams = new URLSearchParams(window.location.search);
+  const selectedVersion = urlParams.get("version") || metadata.version;
```
Comment (anth-volk) [inline src/pages/policy/output/Display.jsx:235]:
```suggestion
  const focus = urlParams.get("focus");
  const selectedVersion = urlParams.get("version") || metadata.version;
```

Comment (anth-volk) [reply src/pages/policy/output/Display.jsx:235]: Without including this, JS interprets the `focus` param in the `BottomCarousel` component as the literal JS function `focus`

File: src/pages/policy/output/ImpactChart.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import HoverCard from "layout/HoverCard";
+import DownloadableScreenshottable from "./DownloadableScreenshottable";
+import { useRef } from "react";
+
+export default function ImpactChart(props) {
+  const { title, description, children } = props;
+  const screenshotRef = useRef();
+  return (
+    <DownloadableScreenshottable ref={screenshotRef}>
+      <h2 style={{ width: "700px", wordWrap: "break-word" }}>{title}</h2>
```
Comment (anth-volk) [inline src/pages/policy/output/ImpactChart.jsx:10]:
```suggestion
      <h2 style={{ width: "100%", wordWrap: "break-word" }}>{title}</h2>
```

Comment (anth-volk) [reply src/pages/policy/output/ImpactChart.jsx:10]: `h2` just continues off the screen without this

**Review by anth-volk — CHANGES_REQUESTED** [review]
Thanks again for this, @abhcs, this seems like a beneficial code refactor to have. I've left a few suggested changes to re-enable the bottom carousel links and prevent a visual bug in the policy output panel.

**Review by anth-volk — APPROVED** [review]
Thanks for your work on this, @abhcs, it's quite a comprehensive refactor. This looks good to me, approving.

**Pull request merged**